### PR TITLE
feat(lambda): python 3.9 runtime

### PR DESCRIPTION
**Pull request by michaelbrewer**
Add Python 3.9 to the list of supported runtimes.

closes: https://github.com/aws/aws-cdk/issues/16144 , https://github.com/aws/aws-cdk/issues/16355

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-lambda/lib/runtime.ts [hunk 1]
```diff
@@ -108,6 +108,14 @@ export class Runtime {
     supportsCodeGuruProfiling: true,
   });
 
+  /**
+   * The Python 3.9 runtime (python3.9)
+   */
+  public static readonly PYTHON_3_9 = new Runtime('python3.9', RuntimeFamily.PYTHON, {
+    supportsInlineCode: true,
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda/lib/runtime.ts:115]: Does it support inline code yet? Usually this support trails behind the actual runtime support.

If so, can you add this to [here](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda/test/integ.runtime.inlinecode.ts) and run the verification documented at the top of the file?

**Review by nija-at — CHANGES_REQUESTED** [review]
Thanks for submitting this PR. See my comment below.

File: packages/@aws-cdk/aws-lambda/lib/runtime.ts (hunk 1 above)
Comment (michaelbrewer) [reply packages/@aws-cdk/aws-lambda/lib/runtime.ts:115]: @nija-at  - support for 3.8 was added
and there is an issue that any version would support inline going forward (https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/80)

But I will add this to the checks.

Comment (michaelbrewer) [reply packages/@aws-cdk/aws-lambda/lib/runtime.ts:115]: @nija-at - added the test and it worked.

**Review by nija-at — APPROVED** [review]
Excellent. Thanks for the update!

**Comment by michaelbrewer** [comment]
Sorry @nija-at i had to move this PR to my personal account, otherwise mergify won't work:

Here is the PR:
- https://github.com/aws/aws-cdk/pull/16366

**Pull request closed without merging**